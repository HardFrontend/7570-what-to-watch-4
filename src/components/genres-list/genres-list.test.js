import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import GenresList from "./genres-list.jsx";

const mockStore = configureStore([]);

const allGenres = [`All genres`, `drama`, `comedy`];
const onFilterClick = () => {};
const films = [
  {
    id: 0,
    title: `Fantastic Beasts`,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `comedy`,
    year: `2011`,
    videosrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 1,
    title: `Bohemian Rhapsody`,
    img: `img/bohemian-rhapsody.jpg`,
    genre: `comedy`,
    year: `2011`,
    videosrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 2,
    title: `Macbeth`,
    img: `img/macbeth.jpg`,
    genre: `comedy`,
    year: `2011`,
    videosrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 3,
    title: `Aviator`,
    img: `img/aviator.jpg`,
    genre: `comedy`,
    year: `2011`,
    videosrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

it(`GenresList render`, () => {
  const store = mockStore({
    sortGenre: `All genres`,
    films,
    filmsShowTo: 8,
    filmsShow: films
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <GenresList
            genres={allGenres}
            onFilterClick={onFilterClick}
          />, </Provider>, {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();

  expect(tree).toMatchSnapshot();
});
