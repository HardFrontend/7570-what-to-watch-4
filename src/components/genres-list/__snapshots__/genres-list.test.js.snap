// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenresList render 1`] = `
Array [
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item
           catalog__genres-item--active"
    >
      <a
        className="catalog__genres-link"
        href="#"
        id="All genres"
        onClick={[Function]}
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item
           "
    >
      <a
        className="catalog__genres-link"
        href="#"
        id="comedy"
        onClick={[Function]}
      >
        comedy
      </a>
    </li>
  </ul>,
  ", ",
]
`;
